specs/WebResourceRequested-CustomScheme.md

```diff
@@ -0,0 +1,295 @@
+// The following XHR will fail because *.example.com is not in the allowed origin list of custom-scheme2.
+// The WebResourceRequested event will not be raised for this request.
+CHECK_FAILURE(m_webView->ExecuteScript(L"var oReq = new XMLHttpRequest();"
+                                       L"oReq.addEventListener(\"load\", reqListener);"
+                                       L"oReq.open(\"GET\", \"custom-scheme-not-in-allowed-origins://example-data.json\");"
+                                       L"oReq.send();",
+                                      Callback<ICoreWebView2ExecuteScriptCompletedHandler>(
+                                          [](HRESULT error, PCWSTR result) -> HRESULT { return S_OK; })
+                                          .Get()));
+```
+
+# API Details
+
+## COM API
+
+```c#
+// This is the ICoreWebView2CustomSchemeRegistration interface
+[uuid(d60ac92c-37a6-4b26-a39e-95cfe59047bb), object, pointer_default(unique)]
+interface ICoreWebView2CustomSchemeRegistration : IUnknown {
+  // Represents the registration of a custom scheme with the CoreWebView2Environment.
+  // This allows the WebView2 app to be able to handle
+  // WebResourceRequested event for requests with the specified scheme and
+  // be able to navigate the WebView2 to the custom scheme. Once the environment
+  // is created, the registrations are valid and immutable throughout the
+  // lifetime of the associated WebView2s' browser process and any WebView2
+  // environments sharing the browser process must be created with identical
+  // custom scheme registrations, otherwise the environment creation will fail.
+  // If there are multiple entries for the same scheme in the registrations list, the environment creation will also fail.
+  // The URIs of registered custom schemes will be treated similar to http URIs for their origins.
+  // They will have tuple origins for URIs with host and opaque origins for
+  // URIs without host as specified in [7.5 Origin - HTML Living Standard](https://html.spec.whatwg.org/multipage/origin.html)
+  // Example:
+  // custom-scheme-with-host://hostname/path/to/resource has origin of custom-scheme-with-host://hostname
+  // custom-scheme-without-host:path/to/resource has origin of custom-scheme-without-host:path/to/resource
+  // For WebResourceRequested event, the cases of request URIs and filter URIs with custom schemes
+  // will be normalized according to generic URI syntax rules. Any non-ASCII characters will be preserved.
+  // The registered custom schemes also participate in [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) and adheres to [CSP](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP). The app needs to set the appropriate access headers in its WebResourceRequested event handler to allow CORS requests.
+
+  // The name of the custom scheme to register.
+  [propget] HRESULT SchemeName([out, retval] LPCWSTR* schemeName);
+  [propput] HRESULT SchemeName([in] LPCWSTR value);
```

Comment:
I don't think this needs to be settable, since it's already set in the constructor.

Reply:
Yes. See also above comments. Just a getter is good.